kong/runloop/handler.lua

```diff
@@ -359,10 +359,20 @@ local function new_router(version)
     end
   end
 
+  -- We need to force detecting router changes if there is some one modifying
+  -- the routers, like rebuild_router_timer.
+  --
+  -- 1. stratey off
+  --      incremental_sync on:
+  --             non init worker: true(kong.core_cache)
+  --                 init worker: false
+  --      incremental_sync off:   false
+  -- 2. strategy on:              true(kong.core_cache)
   local detect_changes = kong.core_cache and true
 
-  -- for dbless we will not check changes when initing
-  if db.strategy == "off" and get_phase() == "init_worker" then
+  if db.strategy == "off" and
```

Comment:
The comment and the code is a little bit odd. Could we change the code to the following which will match the comment:

```suggestion
  detect_changes = detect_changes and (db.strategy ~= "off" or (kong.sync and get_phase() ~= "init_worker"))
```

Reply:
fixed